vllm/v1/engine/async_llm.py

```diff
@@ -140,20 +134,33 @@ def _send_to_streams(self, request_outputs: List[RequestOutput]):
 
         for request_output in request_outputs:
             request_id = request_output.request_id
-            assert request_id in self.request_streams
 
-            self.request_streams[request_id].put(request_output)
+            stream = self.request_streams.get(request_id)
+            if stream is not None:
+                finished = request_output.finished
+                stream.put(request_output)
+                if finished:
+                    self._finish_stream(request_id)
 
-            if request_output.finished:
-                self.request_streams[request_id].finish()
-                self.request_streams.pop(request_id)
+    def _finish_stream(self, request_id: str):
+        stream = self.request_streams.pop(request_id)
+        if stream is not None:
+            stream.finish()
 
-    async def abort_requests(self, request_ids: List[str]) -> None:
+    async def _abort_requests(self, request_ids: List[str]) -> None:
         """Remove request_ids from EngineCore and Detokenizer."""
 
-        if len(request_ids) > 0:
+        # Include any client cancellations.
+        client_aborted_reqs = self.client_aborted_requests
+        if client_aborted_reqs:
+            self.detokenizer.abort_requests(client_aborted_reqs)
+            for request_id in client_aborted_reqs:
+                self._finish_stream(request_id)
+            request_ids.extend(client_aborted_reqs)
+            client_aborted_reqs.clear()
+
+        if request_ids:
             await self.engine_core.abort_requests_async(request_ids)
-            self.detokenizer.abort_requests(request_ids)
```

Comment:
This shouldn't be needed because the detokenizer removes finished requests itself.